accounts/service.go

```diff
@@ -472,7 +468,9 @@ func (s *InterceptorService) invoiceUpdate(
 			s.currentAddIndex, s.currentSettleIndex,
 		)
 		if err != nil {
-			return err
+			return s.disableAndErrorfUnsafe(
```

Comment:
I like this pattern better to be honest. It's more explicit and easier to follow. And one doesn't have to reason about "what comes first, the `defer s.Unlock()` or the `defer func() ... s.disableUnsafe() ...`.

Reply:
Ok, squashed the commits now! Thanks for the feedback :)!